src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/StrandArtifact.java

```diff
@@ -0,0 +1,178 @@
+                .mapToDouble(gaussIntegratorForAlleleFraction::getPoint).boxed().collect(Collectors.toList());
+
+        final GaussIntegrator gaussIntegratorForEpsilon = integratorFactory.legendre(numIntegPointsForEpsilon, 0.0, 1.0);
+        final List<Double> gaussIntegrationWeightsForEpsilon = IntStream.range(0, numIntegPointsForEpsilon)
+                .mapToDouble(gaussIntegratorForEpsilon::getWeight).boxed().collect(Collectors.toList());
+        final List<Double> gaussIntegrationAbscissasForEpsilon = IntStream.range(0, numIntegPointsForEpsilon)
+                .mapToDouble(gaussIntegratorForEpsilon::getPoint).boxed().collect(Collectors.toList());
+
+        double likelihoodForArtifactFwd = 0.0;
+        double likelihoodForArtifactRev = 0.0;
+        double[] forwardLikelihoodsOfAlleleFractions = new double[numIntegPointsForAlleleFraction];
+        double[] reverseLikelihoodsOfAlleleFractions = new double[numIntegPointsForAlleleFraction];
+
+
+        for (int i = 0; i < numIntegPointsForAlleleFraction; i++) {
+            final double f = gaussIntegrationAbscissasForAlleleFraction.get(i);
+            for (int j = 0; j < numIntegPointsForEpsilon; j++) {
+                final double epsilon = gaussIntegrationAbscissasForEpsilon.get(j);
+                final double integrandArtifactFwd = getIntegrandGivenArtifact(f, epsilon, numReadsForward, numReadsReverse, numAltReadsForward, numAltReadsReverse);
+                final double integrandArtifactRev = getIntegrandGivenArtifact(f, epsilon, numReadsReverse, numReadsForward, numAltReadsReverse, numAltReadsForward);
+
+                //
+                likelihoodForArtifactFwd += gaussIntegrationWeightsForAlleleFraction.get(i) * gaussIntegrationWeightsForEpsilon.get(j) * integrandArtifactFwd;
+                likelihoodForArtifactRev += gaussIntegrationWeightsForAlleleFraction.get(i) * gaussIntegrationWeightsForEpsilon.get(j) * integrandArtifactRev;
+
+                // for a fixed f, integrate over epsilons. this gives us the likelihood p(x^+, x^- | f, z) for a fixed f, which is proportional to
+                // the posterior probability of p(f | x^+, x^-, z)
+                forwardLikelihoodsOfAlleleFractions[i] += gaussIntegrationWeightsForEpsilon.get(j) * integrandArtifactFwd;
+                reverseLikelihoodsOfAlleleFractions[i] += gaussIntegrationWeightsForEpsilon.get(j) * integrandArtifactRev;
+            }
+        }
+
+        unnormalized_posterior_probabilities[StrandArtifactZ.ART_FWD.ordinal()] = pi[StrandArtifactZ.ART_FWD.ordinal()] * likelihoodForArtifactFwd;
+        unnormalized_posterior_probabilities[StrandArtifactZ.ART_REV.ordinal()] = pi[StrandArtifactZ.ART_REV.ordinal()] * likelihoodForArtifactRev;
+
+
+        // find the maximum a posteriori estimate for allele fraction given strand artifact
+        final int maxAlleleFractionIndexFwd = MathUtils.maxElementIndex(forwardLikelihoodsOfAlleleFractions);
```

Comment:
I didn't see this comment until after I made the change but nice, you saw that coming. I will look at functions in `OptimizaitonUtils`